### PR TITLE
Modified XML comments for direct method APIs

### PR DESCRIPTION
**Pull request by abhipsaMisra**
~The diffs are highlighted in this commit: https://github.com/Azure/azure-sdk-for-net/pull/11923/commits/3264d2348b91a956b2e7a132c463d0940d85abe9~

Rebased from latest swagger.

### CODE REVIEW

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 1]
```diff
@@ -1653,18 +1653,21 @@
           {
             "name": "deviceId",
             "in": "path",
+            "description": "The Id of the device on whose module the direct method is to be invoked.",
```
Comment (timtay-microsoft) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:1656]: This sentence structure is a bit hard to follow. Maybe "The Id of the device on whose module the direct method will be invoked."

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 2]
```diff
@@ -1,3090 +1,3078 @@
...
+            "description": "Returns the Configuration object",
+            "schema": {
+              "$ref": "#/definitions/Configuration"
+            }
+          }
+        }
+      },
+      "put": {
+        "summary": "Create or update the configuration for devices or modules of an IoT hub. An ETag must not be specified for the create operation. An ETag must be specified for the update operation. Note that configuration Id and Content cannot be updated by the user.",
+        "description": " For IoT Hub VNET related features(https://docs.microsoft.com/en-us/azure/iot-hub/virtual-network-support) please use API version '2020-03-13'.These features are currently in general availability in the East US, West US 2, and Southcentral US regions only. We are actively working to expand the availability of these features to all regions by end of month May. For rest of the APIs please continue using API version '2019-10-01'",
```
Comment (bikamani) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:43]: do we need this?

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 3]
```diff
@@ -2820,22 +2823,24 @@
...
+          "description": "Name of the method which is to be executed.",
           "type": "string"
         },
         "payload": {
-          "description": "Payload",
+          "description": "The JSON-formatted direct method payload, up to 128kb in size.",
           "type": "object"
         },
         "responseTimeoutInSeconds": {
+          "description": "Time (in seconds) that the service waits, for the method invocation to return a response. It defaults to 30 seconds. Minimum allowed value is 5 seconds, maximum allowed value is 300 seconds.",
```
Comment (timtay-microsoft) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:2838]: The comma after "that the service waits," is unnecessary

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 4]
```diff
@@ -2820,22 +2823,24 @@
...
           "format": "int32",
           "type": "integer"
         },
         "connectTimeoutInSeconds": {
+          "description": "Time (in seconds) that the service waits, for the device to come online. It defaults to 0, meaning the device must already be online. Maximum allowed value is 300 seconds.",
```
Comment (timtay-microsoft) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:2843]: Same here. This comma isn't necessary

**Review by timtay-microsoft — APPROVED** [review]
Couple small comments, but otherwise looks good

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 2 above)
Comment (timtay-microsoft) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:43]: The VNet stuff here will be fixed in a different PR. The only reason this is green here is because Abhipsa fixed the spacing of this entire doc

Comment (abhipsaMisra) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:43]: This is a part of the original swagger, which hasn't been cleaned up yet. Look at the commit https://github.com/Azure/azure-sdk-for-net/commit/026cd5902945266a055d940dccaffa42c5f408bb for changes.

**Review by bikamani — APPROVED** [review]

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 5]
```diff
@@ -2834,24 +2834,24 @@
         }
       },
       "CloudToDeviceMethod": {
-        "description": "Parameters to execute a direct method on the device",
+        "description": "Parameters to execute a direct method on the device.",
         "type": "object",
         "properties": {
           "methodName": {
-            "description": "Method to run",
+            "description": "Name of the method which is to be executed.",
```
Comment (vinagesh) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: Name of the method to be executed?

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 6]
```diff
@@ -2834,24 +2834,24 @@
...
             "type": "string"
           },
           "payload": {
-            "description": "Payload",
+            "description": "The JSON-formatted direct method payload, up to 128kb in size.",
             "type": "object"
           },
           "responseTimeoutInSeconds": {
-            "description": "Timeout for the method invocation to return a response",
+            "description": "Timeout for the method invocation to return a response, defaults to 30 seconds. Minimum allowed value is 5 seconds, maximum allowed value is 300 seconds.",
```
Comment (vinagesh) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: Fullstop or semicolon after "return a response" instead of comma

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 7]
```diff
@@ -3054,11 +3054,11 @@
         "properties": {
           "status": {
             "format": "int32",
-            "description": "Method invocation result status.",
+            "description": "Method invocation result status, provided by the device.",
```
Comment (vinagesh) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: Why do we need to say provided by the device? Is the method not invoked on the device?

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 6 above)
Comment (abhipsaMisra) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: This is from an old commit Sindhu, can you review this instead: https://github.com/Azure/azure-sdk-for-net/commit/3264d2348b91a956b2e7a132c463d0940d85abe9

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 7 above)
Comment (abhipsaMisra) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: It is to highlight that this is the status returned by the device, and not the http status.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 8]
```diff
@@ -3054,11 +3054,11 @@
...
             "type": "integer"
           },
           "payload": {
-            "description": "Method invocation result payload.",
+            "description": "The JSON-formatted direct method result payload, up to 128kb in size; provided by the device.",
```
Comment (vinagesh) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: What do you mean by provided by the device? Why do we need to say that? Can this come from anywhere else?

Comment (abhipsaMisra) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: This is to maintain parity with the description for `status` in the result; to highlight that both of these values are set by the device

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 7 above)
Comment (timtay-microsoft) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: To clarify this point, there are two status codes that a user can see when making a device method invocation. The first is the http level status code that service might give if the device is not online, the second is the application level status code that only the device can provide

**Comment by timtay-microsoft** [comment]
I put the swagger spacing fix in my PR, so you can rebase your branch now to simplify what this PR looks like for your reviewers

**Comment by abhipsaMisra** [comment]
> I put the swagger spacing fix in my PR and merged it, so you can rebase your branch now to simplify what this PR looks like for your reviewers

Is your PR merged in? I didn't get a chance to review it yet! :worried:

**Comment by timtay-microsoft** [comment]
> > I put the swagger spacing fix in my PR and merged it, so you can rebase your branch now to simplify what this PR looks like for your reviewers
> 
> Is your PR merged in? I didn't get a chance to review it yet! 😟

Ah, sorry. Priyanka and Binal took a look at it already. I'd be happy to create a new PR if you have any comments. You can still leave those comments on my original PR: https://github.com/Azure/azure-sdk-for-net/pull/11927

**Comment by abhipsaMisra** [comment]
> Ah, sorry. Priyanka and Binal signed off on it already, so I merged it. I'd be happy to create a new PR if you have any comments. You can still leave those comments on my original PR: #11927

No worries, I'll review it now.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 7 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: What do these status look like? Can you give some examples for me to understand?

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 8 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: Trying to understand how this helps from user perspective? Do the SDK users understand difference between device set and application set status?

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 5 above)
Comment (vinagesh) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: Was this changed at all? I think "which is" makes the sentence roundabout. You can just say "Name of the method to be executed on the device"

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 7 above)
Comment (abhipsaMisra) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: This is a sample response:
```csharp
public const string DeviceResponseJson = "{\"name\":\"e2e_test\"}";
new MethodResponse(Encoding.UTF8.GetBytes(DeviceResponseJson), 200)
```
The status here is `200`, but it can be any integer as defined by the application's logic.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 8 above)
Comment (abhipsaMisra) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: For payload in particular the statement `provided by the device` isn't of any significance; since the payload is always set by the device. The status, on the other hand, can be set by either the device or service. (device-set and application-set mean the same thing).

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json (hunk 5 above)
Comment (abhipsaMisra) [reply sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json]: It now says `Name of the method to be executed.`. You are possibly looking at an outdated commit.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 9]
```diff
@@ -2866,24 +2864,24 @@
...
           "type": "string"
         },
         "payload": {
-          "description": "Payload",
+          "description": "The JSON-formatted direct method payload, up to 128kb in size.",
           "type": "object"
         },
         "responseTimeoutInSeconds": {
-          "description": "Timeout for the method invocation to return a response",
+          "description": "Time (in seconds) that the service waits for the method invocation to return a response. It defaults to 30 seconds. Minimum allowed value is 5 seconds, maximum allowed value is 300 seconds.",
```
Comment (vinagesh) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:2879]: Maybe reword? Default is 30 seconds; Minimum allowed value is 5 seconds, Maximum allowed value is 300 seconds.

File: sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json [hunk 10]
```diff
@@ -2866,24 +2864,24 @@
...
           "format": "int32",
           "type": "integer"
         },
         "connectTimeoutInSeconds": {
-          "description": "Timeout for the device to come online",
+          "description": "Time (in seconds) that the service waits for the device to come online. It defaults to 0, meaning the device must already be online. Maximum allowed value is 300 seconds.",
```
Comment (vinagesh) [inline sdk/iot/Azure.Iot.Hub.Service/src/swagger/iothubservice_modified_comments.json:2884]: Maybe reword? Default is 0 (the device must already be online); Maximum allowed value is 300 seconds.

**Review by vinagesh — APPROVED** [review]
